src/Context.ts

```diff
@@ -106,22 +106,35 @@ function getLogger(invocationId: string, functionName: string, log: ILogCallback
     );
 }
 
-function getNormalizedBindingData(request: rpc.InvocationRequest$Properties, info: FunctionInfo): IDict {
+function getNormalizedBindingData(request: rpc.InvocationRequest$Properties): IDict {
   let bindingData: IDict = {
     invocationId: request.invocationId
   };
 
   // node binding data is camel cased due to language convention
   if (request.triggerMetadata) {
-    for (let key in request.triggerMetadata) {
-      let camelCasedKey = key.charAt(0).toLocaleLowerCase() + key.slice(1);
-      bindingData[camelCasedKey] = fromTypedData(request.triggerMetadata[key]);
-    }
+    Object.assign(bindingData, convertKeysToCamelCase(request.triggerMetadata))
   }
 
   return bindingData;
 }
 
+// Recursively convert keys of objects to camel case
+function convertKeysToCamelCase(obj: any) {
+  var output = {};
+  for (var key in obj) {
+      let value = fromTypedData(obj[key]) || obj[key];
+      let camelCasedKey = key.charAt(0).toLocaleLowerCase() + key.slice(1);
+      // If the value is a JSON object, convert keys to camel case

```

Comment:
Can we add a unit test here? Verifying only via E2E tests is getting harder. We should try and improve unit test coverage for each language worker